ridgeback_joint_publisher:
  type: "joint_state_controller/JointStateController"
  publish_rate: 50

turret_position_controller:
  type: "position_controllers/JointTrajectoryController"
  joints:
    - front_camera_pivot_joint
  # type: "position_controllers/JointPositionController"
  # joint: front_camera_pivot_joint
  

ridgeback_velocity_controller:
  type: "mecanum_drive_controller/MecanumDriveController"
  front_left_wheel_joint: "front_left_wheel"
  back_left_wheel_joint: "rear_left_wheel"
  front_right_wheel_joint: "front_right_wheel"
  back_right_wheel_joint: "rear_right_wheel"
  base_frame_id: "base_link"
  odom_frame_id: "odom"
  publish_rate: 50
  pose_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 0.03]
  twist_covariance_diagonal: [0.001, 0.001, 0.001, 1000000.0, 1000000.0, 0.03]
  cmd_vel_timeout: 0.25

  # Override URDF look-up for wheel separation since the parent link is not the base_link.
  wheel_separation_x: 0.638
  wheel_separation_y: 0.551

  # Odometry fused with IMU is published by robot_localization, so
  # no need to publish a TF based on encoders alone.
  enable_odom_tf: false

  # Wheel separation and radius multipliers
  wheel_separation_multiplier: 1.0 # default: 1.0
  wheel_radius_multiplier    : 1.0 # default: 1.0

  # Velocity and acceleration limits
  # Whenever a min_* is unspecified, default to -max_*
  linear:
    x:
      has_velocity_limits    : true
      max_velocity           : 3.0   # m/s
      has_acceleration_limits: true
      max_acceleration       : 5   # m/s^2
    y:
      has_velocity_limits    : true
      max_velocity           : 2.0   # m/s
      has_acceleration_limits: true
      max_acceleration       : 5   # m/s^2
  angular:
    z:
      has_velocity_limits    : true
      max_velocity           : 2.0   # rad/s
      has_acceleration_limits: true
      max_acceleration       : 1.0   # rad/s^2
